### PR TITLE
Add application

### PR DESCRIPTION
**Pull request by yxxhero**
Signed-off-by: yxxhero <aiopsclub@163.com>

## Description
add callsystem

## Related Issue
https://github.com/dragonflyoss/Dragonfly2/issues/678

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist:

- [x] Code compiles correctly.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by cndoit18 — DISMISSED** [review]
`Copyright 2021 The Dragonfly Authors`
协议是不是需要更新一下年份?

File: manager/model/callsystem.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package model
+
+type CallSystem struct {
+	Model
+	Name              string             `gorm:"column:name;type:varchar(256);index:uk_callsystem_name,unique;not null;comment:name" json:"name"`
+	LimitFrequency    string             `gorm:"column:limit_frequency;type:varchar(1024);comment:limit_frequency" json:"limit_frequency"`
```
Comment (gaius-qi) [inline manager/model/callsystem.go:22]: FrequencyLimit

What is the specific value?

File: manager/model/callsystem.go [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+	URLRegexs         Array              `gorm:"column:url_regexs;not null;comment:url_regexs" json:"url_regexs"`
```
Comment (gaius-qi) [inline manager/model/callsystem.go:23]: Why use regexs array?

Isn't regularity matching multiple content?

File: manager/model/callsystem.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	ValidityPeriod    string             `gorm:"column:validity_period;type:varchar(1024);comment:validity_period" json:"validity_period"`
+	IsEnable          bool               `gorm:"column:is_enable;not null;default:true;comment:enable callsystem" json:"is_enable"`
```
Comment (gaius-qi) [inline manager/model/callsystem.go:25]: Use `State` and set `enable` or `disable`. Refer to `User` model.

File: manager/model/callsystem.go [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+package model
+
+type CallSystem struct {
+	Model
+	Name              string             `gorm:"column:name;type:varchar(256);index:uk_callsystem_name,unique;not null;comment:name" json:"name"`
+	LimitFrequency    string             `gorm:"column:limit_frequency;type:varchar(1024);comment:limit_frequency" json:"limit_frequency"`
+	URLRegexs         Array              `gorm:"column:url_regexs;not null;comment:url_regexs" json:"url_regexs"`
+	ValidityPeriod    string             `gorm:"column:validity_period;type:varchar(1024);comment:validity_period" json:"validity_period"`
```
Comment (gaius-qi) [inline manager/model/callsystem.go:24]: Remove `ValidityPeriod`. Let the user switch the `State` of enable or disable.

File: manager/model/callsystem.go [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+	IsEnable          bool               `gorm:"column:is_enable;not null;default:true;comment:enable callsystem" json:"is_enable"`
+	SchedulerClusters []SchedulerCluster `json:"-"`
```
Comment (gaius-qi) [inline manager/model/callsystem.go:26]: Add CDNClusters.

File: manager/types/callsystem.go [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+	LimitFrequency string   `json:"limit_frequency" binding:"required"`
+	URLRegexs      []string `json:"url_regexs" binding:"required"`
+	ValidityPeriod string   `json:"validity_period" binding:"required"`
+	IsEnable       bool     `json:"is_enable" binding:"omitempty"`
+}
+
+type UpdateCallSystemRequest struct {
+	Name           string   `json:"name" binding:"required"`
+	LimitFrequency string   `json:"limit_frequency" binding:"required"`
+	URLRegexs      []string `json:"url_regexs" binding:"required"`
```
Comment (gaius-qi) [inline manager/types/callsystem.go:44]: omitempty

File: manager/types/callsystem.go [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+	ValidityPeriod string   `json:"validity_period" binding:"required"`
+	IsEnable       bool     `json:"is_enable" binding:"omitempty"`
+}
+
+type UpdateCallSystemRequest struct {
+	Name           string   `json:"name" binding:"required"`
+	LimitFrequency string   `json:"limit_frequency" binding:"required"`
```
Comment (gaius-qi) [inline manager/types/callsystem.go:43]: omitempty

**Review by gaius-qi — CHANGES_REQUESTED** [review]
Change the name of `CallSystem`

File: manager/model/callsystem.go (hunk 2 above)
Comment (yxxhero) [reply manager/model/callsystem.go:23]: I think  you are right.

File: manager/model/application.go [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package model
+
+type Application struct {
+	Model
+	Name              string             `gorm:"column:name;type:varchar(256);index:uk_callsystem_name,unique;not null;comment:name" json:"name"`
```
Comment (gaius-qi) [inline manager/model/application.go:21]: index is uk_application_name

**Review by gaius-qi — CHANGES_REQUESTED** [review]

**Review by gaius-qi — APPROVED** [review]
LGTM

**Review by jim3ma — APPROVED** [review]

**Pull request merged**